File: Backend/BaseMgtService/Senior.AgileAI.BaseMgt.Application/DTOs/UpdateRequirementsDTO.cs
using Senior.AgileAI.BaseMgt.Domain.Enums;

namespace Senior.AgileAI.BaseMgt.Application.DTOs
{
    public class UpdateRequirementsDTO
    {
        public required Guid RequirementId { get; set; }
        public string? Title { get; set; }
        public string? Description { get; set; }
        public RequirementsStatus? Status { get; set; }
        public ReqPriority? Priority { get; set; }

    }
}

File: agilemeets_mobile/lib/data/models/requirements/update_requirements_dto.dart
import '../../enums/req_priority.dart';
import '../../enums/requirements_status.dart';

class UpdateRequirementsDTO {
  final String requirementId;
  final String title;
  final String? description;
  final RequirementStatus status;
  final ReqPriority priority;

  const UpdateRequirementsDTO({
    required this.requirementId,
    required this.title,
    this.description,
    required this.status,
    required this.priority,
  });

  Map<String, dynamic> toJson() => {
    'requirementId': requirementId,
    'title': title,
    'description': description,
    'status': status.value,
    'priority': priority.value,
  };
} 